### PR TITLE
Fix linking issues by fixing LIBRARY_SEARCH_PATHS.

### LINKED ISSUE #6: iOS app doesn't build
I followed the installation steps and when running `yarn ios-ui` I get the following error:

```
❌  ld: library not found for -lappcenter-analytics

❌  clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I notice that CI (GitHub actions) is only running for the Android build. It should be running for iOS too.

### PR DESCRIPTION
**Pull request by mluisbrown**
Fixes #6 

The `LIBRARY_SEARCH_PATHS` had extra libraries that are not used and was missing libraries that are required. This PR tidies them up. 

It should not actually be necessary to set the search paths by hand, as this should come automatically from the Cocoapods `.xcconfig` files, however, these include the `$(inherited)` path which for some reason causes a strange linker failure as described [here](https://developer.apple.com/forums/thread/133503). This seems to be an issue related to ReactNative, as all the occurrences of this issue mention RN, so for now I guess maintaining `LIBRARY_SEARCH_PATHS` by hand is the only way.

### CODE REVIEW

File: ios/Podfile [hunk 1]
```diff
@@ -8,7 +8,7 @@
 # SPDX-License-Identifier: EUPL-1.2
 #
 
-platform :ios, '11.0'
+platform :ios, '13.5'
 require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
 source 'https://github.com/CocoaPods/Specs.git'
```
Comment (djcouto) [inline ios/Podfile:13]: Could you please remove this unnecessary sources too?

lines #13 and #14

Comment (mluisbrown) [reply ios/Podfile]: I've removed the sources lines.

**Comment by mluisbrown** [comment]
Hi @djcouto , I tried your suggestion but with that I get the same linker error that I mentioned before:

```
Undefined symbols for architecture x86_64:
  "_swift_getTypeByMangledNameInContextInMetadataState", referenced from:
      ___swift_instantiateConcreteTypeFromMangledNameAbstract in Backend.o
      ___swift_instantiateConcreteTypeFromMangledNameAbstract in DP3TSDK.o
ld: symbol(s) not found for architecture x86_64
```

I can only avoid this error if I don't have the `$(inherited)` line in `LIBRARY_SEARCH_PATHS`, and without this line, all the Cocoapods library paths have to be added by hand.

I'm using Xcode 11.6.

**Comment by mluisbrown** [comment]
image

image

**Comment by djcouto** [comment]
@mluisbrown sorry for the delay, I'm trying to reproduce your problem, and until now I couldn't. 

Let me dig into this a little more in other to find a definite solution 👍

**Comment by djcouto** [comment]
@mluisbrown I was able to reproduce the problem, I was dealing with a cache problem.

I believe this problem of linker has something to do with the change that I asked to do on the `Podfile`. The error is related to compiling the DP3T-ios-sdk.

Can you please change your `LIBRARY_SEARCH_PATHS` to:
```
$(inherited)
"$(SRCROOT)/$(TARGET_NAME)"
```

and add back `source 'https://cdn.cocoapods.org/'` to your Podfile, and then run `pod install`

If the problem persists we will merge this PR, since with your approach is working, and move this issue a to a new issue.

**Comment by mluisbrown** [comment]
@djcouto I tried your suggestion but it didn't make any difference. I'm not convinced that the problem is related to the DP3T-ios-sdk, even though the error is reported when linking `DPT3SDK.o`. I think it's a more general problem that has something to do with React Native. 

The only places where I find this error mentioned on the web are [here](https://developer.apple.com/forums/thread/133503) and [here](https://github.com/fastlane/fastlane/issues/16277) and both are for ReactNative apps.

The only difference to what is "generated" for `LIBRARY_SEARCH_PATHS` when using `$(inherited)` or not is the first 2 "toolchain" entries. The problem lies there somewhere, but it's somehow connected with React Native.

image

So the problem persists and IMO we should merge this to have something building. We can revisit this later if we find a better solution.

**Comment by mluisbrown** [comment]
I've created a post on the Swift Forums about this to see if anyone there has any ideas: https://forums.swift.org/t/strange-linker-error-mentioning-swift-gettypebymanglednameincontextinmetadatastate/38936

**Comment by djcouto** [comment]
@mluisbrown I agree we can revisit this later if we find a better solution. For now, let's merge this.

Thank you for your availability 👍

**Pull request merged**